Pause button for buildings

Heldereim opened the pull request:
Added a button in building info selling tab that when pressed pauses the operations of a building. Also fixed a bug where wood cutters and other buildings with byproducts would keep operating even though it had a full inventory of byproducts if it was set to selling.

Heldereim committed:
Pause button for buildings

Added a button in building info selling tab that when pressed pauses the operations of a building. Also fixed a bug where wood cutters and other buildings with byproducts would keep operating even though it had a full inventory of byproducts if it was set to selling.

Heldereim committed:
Buildinginfo hidden by default

linuskar commented on scripts/buildings/gathering_building.gd:
```diff
@@ -28,36 +28,17 @@ func check_if_can_produce() -> bool:
-## resources in output.
-func check_for_output_overflow() -> bool:
-	for produced_good in produced_goods:
-		var produced_good_generated: int = output_generation.get(produced_good) * resource_tiles_to_gather.size()
```
Need to include the amount of resource tiles a gathering building is gathering for overflow to work properly. So check_for_production_overflow() needs to be overrided with by multiplying resource_tiles_to_gather.size() to the produced goods generated.

linuskar commented on scripts/UI/BuildingInfo.gd:
```diff
@@ -343,3 +343,9 @@ func clean_buildings_list(buildings: Array[Building]) -> Array[Building]:
 			cleaned_list.set(current, true)
 	return cleaned_list.keys()
+
+func pause_production() -> void:
```
Need to set the sell status label when setting buildings active. As the text wont change from building to building. For example the label will still be paused when switching from to another building. I tried to fix this in the branch for research costs by setting the label when building info ui is set active.
![image](https://github.com/user-attachments/assets/9211d02e-5f41-44e6-bd01-f0d9bfd7d1e6)

Heldereim replied:
Solution exists in latest commit

linuskar commented on scripts/buildings/production_building.gd:
```diff
@@ -99,17 +99,40 @@ func _handle_produced_goods() -> void:
 ## Function to check if the production building can produce.
 func check_if_can_produce() -> bool:
-	var missing_input: bool = check_for_missing_input()
 	var can_be_output_overflow: bool = check_for_output_overflow()
```
remove row 102 if not used.

linuskar commented on scripts/buildings/production_building.gd:
```diff
@@ -218,3 +241,13 @@ func check_restart_production() -> void:
+## Function for pausing smoke emissions
+func pause_smokes() -> void:
+	for smoke in smokes:
+			smoke.emitting = false
```
One too many indents

Heldereim committed:
Mode updates for buildings

Made some changes so that the building info tab showing the mode of the building selected now shows 'Mixed' when selecteing multiple buildings with different modes. Added the ProductionBuildingMode enum to the Enums script. Added text interpretations of the modes to Enums as well as a function for getting these interpretations. Made it so that production buildings no longer use a bool for checking what mode they are in but instead use the ProductionBuildingMode enum.

Heldereim committed:
Overrode check_for_production_overflow in gathering building

Overrode the function check_for_production_overflow in gathering building to account for gathering buildings of larger size than 1x1

linuskar commented on scripts/UI/BuildingInfo.gd:
```diff
@@ -343,3 +342,25 @@ func clean_buildings_list(buildings: Array[Building]) -> Array[Building]:
+func set_building_storing() -> void:
+	set_building_mode(Enums.ProductionBuildingMode.STORING)
+
+func set_mode_label() -> void:
```
You may want to update from main, as i had implemented something similar before. So probably just remove the previous function and its usage in set active?

Heldereim committed:
Merge branch 'main' into 9-factories-should-have-an-off-button

Heldereim committed:
Code removed

Removed code sourced from main whose function was duplicated

linuskar reviewed (approved):
LGTM
